packages/react-positioning/src/usePopper.ts

```diff
@@ -1,65 +1,40 @@
+import { useEventCallback } from '@fluentui/react-utilities';
 import * as React from 'react';
-
-import { isIntersectingModifier } from './isIntersectingModifier';
+import { PopperOptions, PositioningProps, PopperVirtualElement } from './types';
+import { getPlacement } from './utils/positioningHelper';
+import { useCallbackRef } from './utils/useCallbackRef';
 import {
-  getScrollParent,
-  applyRtlToOffset,
-  getPlacement,
-  getReactFiberFromNode,
-  getBoundary,
-  useCallbackRef,
-  getBasePlacement,
-} from './utils/index';
-import type { PopperVirtualElement, PopperOptions, PositioningProps } from './types';
-
-type PopperInstance = PopperJs.Instance & { isFirstRun?: boolean };
+  offset as offsetMiddleware,
+  shift as shiftMiddleware,
+  flip as flipMiddleware,
+  coverTarget as coverTargetMiddleware,
+  maxSize as maxSizeMiddleware,
+  intersectionObserver as intersectionObserverMiddleware,
+} from './middleware/index';
+import { getScrollParent } from './utils/getScrollParent';
+import debounce from './utils/debounce';
+import { useFirstMount } from '@fluentui/react-utilities';
+import { dataPopperEscaped, dataPopperIntersecting, dataPopperReferenceHidden } from './contants';
+import { addEventListenerOnScrollParent } from './utils/addEventListenerOnScrollParent';
+import { hasAutofocusFilter } from './utils/hasAutoFocusFilter';
 
 interface UsePopperOptions extends PositioningProps {
   /**
-   * If false, delays Popper's creation.
-   * @default true
+   * If false, does not position anything
    */
   enabled?: boolean;
 }
 
-//
-// Dev utils to detect if nodes have "autoFocus" props.
-//
-
-/**
- * Detects if a passed HTML node has "autoFocus" prop on a React's fiber. Is needed as React handles autofocus behavior
- * in React DOM and will not pass "autoFocus" to an actual HTML.
- */
-function hasAutofocusProp(node: Node): boolean {
-  // https://github.com/facebook/react/blob/848bb2426e44606e0a55dfe44c7b3ece33772485/packages/react-dom/src/client/ReactDOMHostConfig.js#L157-L166
-  const isAutoFocusableElement =
-    node.nodeName === 'BUTTON' ||
-    node.nodeName === 'INPUT' ||
-    node.nodeName === 'SELECT' ||
-    node.nodeName === 'TEXTAREA';
-
-  if (isAutoFocusableElement) {
-    return !!getReactFiberFromNode(node)?.pendingProps.autoFocus;
-  }
-
-  return false;
-}
-
-function hasAutofocusFilter(node: Node) {
-  return hasAutofocusProp(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
-}
-
-/**
- * Provides a callback to resolve Popper options, it's stable and should be used as a dependency to trigger updates
- * of Popper options.
- *
- * A callback is used there intentionally as some of Popper.js modifiers require DOM nodes (targer, container, arrow)
- * that can't be resolved properly during an initial rendering.
- */
-function usePopperOptions(options: PopperOptions, popperOriginalPositionRef: React.MutableRefObject<string>) {
+function usePopperOptions(options: PopperOptions) {
```

Comment:
As this function is not exported, is there sense to rename it? `useFloatingUIOptions`?